src/components/calcite-popover/calcite-popover.tsx

```diff
@@ -0,0 +1,160 @@
+import Popper from "popper.js";
+
+@Component({
+  tag: "calcite-popover",
+  styleUrl: "calcite-popover.scss",
+  shadow: true
+})
+export class CalcitePopover {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Determines where the element will be positioned.
+   * horizontal: Positioned to the left or right of the positionElement.
+   * vertical: Positioned above or below the positionElement.
+   */
+  @Prop({ reflect: true }) placement: CalcitePlacement = "horizontal";
+
+  @Watch("placement")
+  placementHandler() {
+    this.reposition();
+  }
+
+  /**
+   * Reference HTMLElement used to position this component according to the placement property.
+   */
+  @Prop() positionElement: HTMLElement;
+
+  @Watch("positionElement")
+  positionElementHandler() {
+    this.destroyPopper();
+    this.reposition();
+  }
+
+  /**
+   * Offset the position of the popover in the horizontal direction.
+   */
+  @Prop({ reflect: true }) xOffset = 0;
+
+  @Watch("xOffset")
+  xOffsetHandler() {
+    this.reposition();
+  }
+
+  /**
+   * Offset the position of the popover in the vertical direction.
+   */
+  @Prop({ reflect: true }) yOffset = 0;
+
+  @Watch("yOffset")
+  yOffsetHandler() {
+    this.reposition();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteShellFloatingPanelElement;
+
+  @State() popper: Popper;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  componentDidLoad() {
+    this.reposition();
+  }
+
+  componentDidUnload() {
+    this.destroyPopper();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async reposition(): Promise<void> {
+    const { el, placement: componentPlacement, popper, positionElement, xOffset, yOffset } = this;
+
+    const placement = componentPlacement === "vertical" ? "bottom-start" : "auto-start";
+    const offsetEnabled = !!(yOffset || xOffset);
+
+    const modifiers = {
+      offset: {
+        enabled: offsetEnabled,
+        offset: `${yOffset}, ${xOffset}`
+      },
+      preventOverflow: {
+        enabled: false
+      }
+    };
+
+    if (popper) {
+      popper.options.placement = placement;
+      popper.options.modifiers = { ...popper.options.modifiers, ...modifiers };
+      popper.scheduleUpdate();
+      return;
+    }
+
+    if (el && placement && positionElement) {
+      const newPopper = new Popper(positionElement, el, {
```

Comment:
the docs are somewhat unclear, but you may want to enable `removeOnDestroy`. Destroying it doesn't remove it from the dom by default?

Reply:
The component will handling removing the DOM.

Reply:
I read about `removeOnDestroy` in the Popper.js options docs. It's false by default. You know what it does (or doesn't do when disabled)? Seems like something we might want but its hard to tell. You sure it removes it from the dom with this being false? If so then what's this for?

Reply:
Yeah, we don't need it. Its for someone who wants to remove the node when they call destroy on the popper. We don't need that because the component handles it.

Reply:
https://github.com/FezVrasta/popper.js/blob/f41c861582ee75ccab44b2e3467c4ec63399dbfa/packages/popper/src/methods/destroy.js#L29

Reply:
That article shows using a throttle for resize. Likewise, it states that

> Currently, only touchstart, touchmove and mousewheel event handlers could be made passive

Also, passive isn't supported on IE11. https://caniuse.com/#feat=passive-event-listener

Reply:
@pr3tori4n We don't support IE11. The function that gets called is already throttled.